pkgs/build-support/trivial-builders.nix

```diff
@@ -46,6 +46,18 @@ rec {
   writeScript = name: text: writeTextFile {inherit name text; executable = true;};
   writeScriptBin = name: text: writeTextFile {inherit name text; executable = true; destination = "/bin/${name}";};
 
+  # Create a Shell script, check its syntax
+  writeShellScriptBin = name : text :
+    let
+      s = writeScriptBin name ''
+        #!${stdenv.shell}
+        ${text}
+      '';
+    in
+      runCommand name {} ''
+        ${stdenv.shell} -n ${s}/bin/${name}
+        cp -v -r ${s} $out
```

Comment:
I think this change is great!

Still, instead of making a copy of the verified script, I think it would be better to add a checkPhase to the writeTextFile function above.